### PR TITLE
Simplify Splash Screen Launch for integration call

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Simplify Splash Screen Launch for integration call.

```
splashScreen = new Dynamo.UI.Views.SplashScreen();
splashScreen.webView.NavigationCompleted += LoadDynamoView;
splashScreen.Show();

LoadDynamoView()
{
       // Initializing DynamoMolde, and DynamoViewModel
       ...
       splashScreen.DynamoView = new DynamoView(viewModel);
       splashScreen.OnRequestStaticSplashScreen();
       splashScreen.webView.NavigationCompleted -= LoadDynamoView;
}
```

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

N/A

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 1]
```diff
@@ -23,30 +29,172 @@ public partial class SplashScreen : Window
...
+            if (webView != null)
+            {
+                webView.NavigationCompleted -= WebView_NavigationCompleted;
+            }
+        }
+
+        private void LoadDynamoView()
+        {
+            DynamoModel model;
+            model = StartupUtils.MakeModel(false, ASMPath ?? string.Empty, hostAnalyticsInfo);
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:93]: The DynamoModel and DynamoViewModel currently are both hard coded to the one starting with default config here. This is fine for Sandbox but will be different in other cases, I would like to brainstorm with you to figure out how to pass in DynamoModel and DynamoViewModel and make things happen async. @RobertGlobant20 @filipeotero let's meet later

File: src/DynamoSandbox/DynamoCoreSetup.cs [hunk 2]
```diff
@@ -145,138 +127,9 @@ public void RunApplication(Application app)
...
-        }
-
-        private void CloseMigrationWindow()
-        {
-            if (splashScreen == null)
-                return;
-
-            splashScreen.Close();
-            splashScreen = null;
-        }
```
Comment (QilongTang) [inline src/DynamoSandbox/DynamoCoreSetup.cs:280]: These two functions no longer used, so removing

File: src/DynamoSandbox/DynamoCoreSetup.cs [hunk 3]
```diff
@@ -145,138 +127,9 @@ public void RunApplication(Application app)
...
-
         private void ASMPreloadFailureHandler(string failureMessage)
         {
             MessageBoxService.Show(failureMessage, "DynamoSandbox", MessageBoxButton.OK, MessageBoxImage.Warning);
         }
-
-        void OnDynamoViewLoaded(object sender, RoutedEventArgs e)
-        {
-            CloseMigrationWindow();
-            Analytics.TrackStartupTime("DynamoSandbox", startupTimer.Elapsed);
```
Comment (QilongTang) [inline src/DynamoSandbox/DynamoCoreSetup.cs:270]: This is now moved to SplashScreen code

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 4]
```diff
@@ -23,30 +29,172 @@ public partial class SplashScreen : Window
...
+        /// <summary>
+        /// Dynamo View Model reference
+        /// </summary>
+        public DynamoViewModel viewModel;
 
         /// <summary>
         /// Constructor
         /// </summary>
-        public SplashScreen(string userDataDirectory)
+        public SplashScreen(HostAnalyticsInfo info = new HostAnalyticsInfo(), string asmPath ="", string cerPath ="", string cmdFilePath ="")
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:54]: The signature here should be updated later

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 5]
```diff
@@ -19,34 +23,183 @@ public partial class SplashScreen : Window
...
+        /// Handler to launch Dynamo View
+        /// </summary>
+        /// <param name="isCheckboxChecked"></param>
+        private void LaunchDynamo(bool isCheckboxChecked)
+        {
+            viewModel.PreferenceSettings.EnableStaticSplashScreen = !isCheckboxChecked;
+            DynamoModel.RequestUpdateLoadBarStatus -= DynamoModel_RequestUpdateLoadBarStatus;
+            StaticSplashScreenReady -= FinishLoadingDynamicScreen;
+            Close();
+            Application.Current.MainWindow = dynamoView;
```
Comment (RobertGlobant20) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:164]: I guess this line will change if we plan to show the SplashScreen from Revit when Dynamo is starting.

File: src/DynamoSandbox/DynamoCoreSetup.cs [hunk 6]
```diff
@@ -62,34 +60,23 @@ public void RunApplication(Application app)
         {
             try
             {
-                //This line validates if the WebView2 Runtime is installed in the computer, if is not we return and then exit Dynamo
+                // This line validates if the WebView2 Runtime is installed in the computer before launching DynamoSandbox,
+                // if is not we return and then exit Dynamo Sandbox
                 if (!WebView2Utilities.ValidateWebView2RuntimeInstalled())
```
Comment (RobertGlobant20) [inline src/DynamoSandbox/DynamoCoreSetup.cs:65]: Not sure If we are planning to show the message saying the WebView2 is not installed when running Dynamo over Revit but in that case we should move also this code to SplashScreen.xaml.cs

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 7]
```diff
@@ -19,34 +23,183 @@ public partial class SplashScreen : Window
...
+        /// </summary>
+        public DynamoView DynamoView
+        {
+            get
+            {
+                return dynamoView;
+            }
+            set
+            {
+                dynamoView = value;
```
Comment (RobertGlobant20) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:58]: is this variable set in any place right now? or is just preparation for when the code is called from DynamoRevit?

**Review by RobertGlobant20 — COMMENTED** [review]
LGTM with some comments

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs (hunk 7 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:58]: This setter is currently called in CoreSetupfile

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs (hunk 5 above)
Comment (QilongTang) [reply src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:164]: This line will not change, just the variable of `dynamoView` will point to the view initialized in DynamoRevit

File: src/DynamoSandbox/DynamoCoreSetup.cs [hunk 8]
```diff
@@ -225,58 +152,15 @@ private void LoadDynamoView()
                    });
 
             DynamoModel.OnRequestUpdateLoadBarStatus(new SplashScreenLoadEventArgs(Dynamo.Wpf.Properties.Resources.SplashScreenLaunchingDynamo, 70));
-            dynamoView = new DynamoView(viewModel);
-            authManager = model.AuthenticationManager;
+            splashScreen.DynamoView = new DynamoView(viewModel);
```
Comment (QilongTang) [inline src/DynamoSandbox/DynamoCoreSetup.cs:155]: @RobertGlobant20 the setter is used here

**Comment by QilongTang** [comment]
@filipeotero @zeusongit @reddyashish Any more comments before merging?

**Review by zeusongit — APPROVED** [review]
LGTM

**Pull request merged**